Foundation/NSStream.swift

```diff
@@ -154,17 +205,97 @@ public class InputStream: Stream {
         CFReadStreamClose(_stream)
     }
     
+    public override var streamError: NSError?{
+        let error = CFReadStreamCopyError(_stream)
+        return error?._nsObject
+    }
+    
     public override var streamStatus: Status {
-        return Stream.Status(rawValue: UInt(CFReadStreamGetStatus(_stream)))!
+        return NSStream.Status(rawValue: UInt(CFReadStreamGetStatus(_stream)))!
+    }
+    
+    public override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        return CFReadStreamSetProperty(_stream, key._cfObject, property)
+    }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFReadStreamCopyProperty(_stream, key._cfObject)
     }
+    
+    public func copy(with zone: NSZone?) -> AnyObject{
+        if let data = _data {
+            return NSInputStream(data:data)
+        } else if let url = _url {
+            return NSInputStream(url:url)!
+        } else {
+          NSRequiresConcreteImplementation()
+        }
+    }
+    
+    public func mutableCopy(with zone: NSZone?) -> AnyObject{
+        return copy(with: nil)
+    }
+
+}
+
+internal final class _SwiftOutputStream : NSOutputStream, _SwiftNativeFoundationType {
+    internal typealias ImmutableType = NSOutputStream
+    internal typealias MutableType = NSOutputStream
```

Comment:
not sure how to satisfy _SwiftNativeFoundationType with out making the MutableType to OutputStream